packages/perseus/src/widgets/phet-sim.tsx

```diff
@@ -0,0 +1,259 @@
+/* eslint-disable react/forbid-prop-types, react/sort-comp */
+/**
+ * This is a PhET simulation widget. It is used for rendering simulations
+ * from https://phet.colorado.edu/.
+ */
+
+import Banner from "@khanacademy/wonder-blocks-banner";
+import {View} from "@khanacademy/wonder-blocks-core";
+import IconButton from "@khanacademy/wonder-blocks-icon-button";
+import cornersOutIcon from "@phosphor-icons/core/regular/corners-out.svg";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+
+import {PerseusI18nContext} from "../components/i18n-context";
+import {getDependencies} from "../dependencies";
+import * as Changeable from "../mixins/changeable";
+import {phoneMargin} from "../styles/constants";
+import {basicBorderColor, borderRadiusLarge} from "../styles/global-constants";
+
+import type {PerseusPhetSimWidgetOptions} from "../perseus-types";
+import type {WidgetExports, WidgetProps} from "../types";
+
+type RenderProps = PerseusPhetSimWidgetOptions; // transform = _.identity
+type Props = WidgetProps<RenderProps, PerseusPhetSimWidgetOptions>;
+
+// For returning user input, but currently the PhET widget
+// does not support accessing user input
+type UserInput = null;
+
+type State = {
+    banner: {
+        message: string;
+        kind: "warning" | "critical";
+    } | null;
+    url: URL | null;
+};
+
+/* This renders the PhET sim */
+export class PhetSim extends React.Component<Props, State> {
+    static contextType = PerseusI18nContext;
+    declare context: React.ContextType<typeof PerseusI18nContext>;
+    private readonly iframeRef: React.RefObject<HTMLIFrameElement> =
+        React.createRef<HTMLIFrameElement>();
+    private readonly locale: string;
+
+    state: State = {
+        url: null,
+        banner: null,
+    };
+
+    constructor(props) {
+        super(props);
+        this.locale = this.getPhetCompatibleLocale(getDependencies().kaLocale);
+    }
+
+    getUserInput(): UserInput {
+        return null;
+    }
+
+    async componentDidMount() {
+        await this.updateSimState(this.props.url);
+    }
+
+    async componentDidUpdate(prevProps) {
+        // If the URL has changed, update our state
+        if (prevProps.url !== this.props.url) {
+            await this.updateSimState(this.props.url);
+        }
+    }
+
+    render(): React.ReactNode {
+        // We sandbox the iframe so that we allowlist only the functionality
+        // that we need. This makes it safer to present third-party content
+        // from the PhET website.
+        // http://www.html5rocks.com/en/tutorials/security/sandboxed-iframes/
+        const sandboxProperties = "allow-same-origin allow-scripts";
+        return (
+            <View style={styles.container}>
+                {this.state.banner !== null && (
+                    // TODO(anna): Make this banner focusable
```

Comment:
Suggestion, nb: Might be good to have this TODO right above the banner you want to make focusable.